components/kubeflow-pipeline/pipeline_components/analytics-client/Makefile

```
@@ -5,18 +5,23 @@ MAKEVAR_REGISTRY?=ghcr.io/katulu-io/fl-suite
 CONTAINER_TAG = $(shell echo "${MAKEVAR_VERSION}" | sed "s/\+[0-9a-zA-Z-]*//")
 
 lint:
-	@echo "lint not implemented"
+	poetry install --no-root
```

Consider factoring out the `poetry install --no-root` into it's own target, e.g:

```suggestion
POETRY_STAMP := .poetry.stamp

${POETRY_STAMP}: backend/poetry.lock
	@poetry install --no-root
	@touch ${POETRY_STAMP}

lint: ${POETRY_STAMP}
```

This will basically only run the `poetry install` when there are changes in the `poetry.lock`, e.g when a new dependency is added in the repo and locally `poetry install` has not been run. A similar implementation can be seen inside the `edge-identity` component.

---

That would be great, unfortunately with Make we can only check if files exist, not that they have been changed.